Reqnroll/Formatters/ExecutionTracking/TestRunHookExecutionTracker.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Io.Cucumber.Messages.Types;
+using Reqnroll.Formatters.PayloadProcessing.Cucumber;
+using Reqnroll.Events;
+using System;
+using System.Collections.Generic;
+
+namespace Reqnroll.Formatters.ExecutionTracking;
+
+/// <summary>
+/// Captures information about TestRun Hooks (Before/After TestRun and Before/After Feature)
+/// </summary>
+public class TestRunHookExecutionTracker(string hookStartedId, string hookId, string testRunId, ICucumberMessageFactory messageFactory)
+    : IGenerateMessage
+{
+    public string TestRunId { get; } = testRunId;
+    public string HookId { get; } = hookId;
+    public string HookStartedId { get; } = hookStartedId;
+
+    public DateTime HookStarted { get; private set; }
```

Comment:
Consider `DateTime?` to avoid comparisons with `default`.

Reply:
Completed.